go/pkg/filemetadata/filemetadata_test.go

```diff
@@ -118,6 +124,87 @@ func TestComputeFilesWithXattr(t *testing.T) {
 	}
 }
 
+func TestComputeFileDigestWithXattr(t *testing.T) {
+	xattrDgName := "user.myhash"
+	overwriteXattrDgName(t, xattrDgName)
+	tests := []struct {
+		name       string
+		contents   string
+		xattrDgStr string
+		wantDgStr  string
+		wantErr    bool
+	}{
+		{
+			name:      "no-xattr",
+			contents:  "123456",
+			wantDgStr: "8d969eef6ecad3c29a3a629280e686cf0c3f5d5a86aff3ca12020c923adc6c92/6",
+		},
+		{
+			name:       "only-digest-hash",
+			contents:   "123456",
+			xattrDgStr: "1111111111111111111111111111111111111111111111111111111111111111",
+			wantDgStr:  "1111111111111111111111111111111111111111111111111111111111111111/6",
+		},
+		{
+			name:       "full-digest-(hash+size)",
+			contents:   "",
+			xattrDgStr: "1111111111111111111111111111111111111111111111111111111111111111/666",
+			wantDgStr:  "1111111111111111111111111111111111111111111111111111111111111111/666",
+		},
+		{
+			name:       "invalid-digest-hash",
+			contents:   "123456",
+			xattrDgStr: "abc",
+			wantDgStr:  "abc/6",
+			wantErr:    true,
+		},
+		{
+			name:       "invalid-full-digest",
+			contents:   "123456",
+			xattrDgStr: "666/666",
+			wantDgStr:  "666/666",
+			wantErr:    true,
+		},
+		{
+			name:       "invalid-full-digest-(extra-slash)",
+			contents:   "123456",
+			xattrDgStr: "///666",
+			wantDgStr:  "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/0",
+			wantErr:    true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			filePath := tc.name
```

Comment:
nit: Does the file path need to be the test name? That adds the constraint that the test name can be a valid file path. You can just write to a constant or random file path.

Reply:
changed to a const named "test.txt"